deployer/scripts/upgrade.sh

```diff
@@ -278,20 +286,103 @@ function getArrayIndex() {
   fi
 }
 
+# This function will wait until the latestVersion changes
+function waitForChange() {
+
+  local currentVersion=$1
+  local dc=$2
+
+  waitFor "[[ $currentVersion -lt \$(oc get $dc -o jsonpath='{.status.latestVersion}') ]]" && return 0
+
+  return 1
+}
+
+# This function will tell us if the proposed patches are different than the current values
+function isValidChange() {
+
+  local object=$1
+  local patches=$2
+
+  for patch in $patches; do
```

Comment:
I don't like using a single parameter for multiple things like this, and I think it would be just as clear and more correct (if you ever had a patch with a space) as:

```
function isValidChange() {

  local object=$1; shift

  for patch in "$@"; do
  ...
}

isValidChange "$object" "${patches[@]}"